docs/configuration/config-file-reference.md

```diff
@@ -4097,6 +4097,17 @@ The `limits_config` configures default and per-tenant limits imposed by Cortex s
 # override is set, the encryption context will not be provided to S3. Ignored if
 # the SSE type override is not set.
 [s3_sse_kms_encryption_context: <string> | default = ""]
+
+# Per-user rate limit for sending email notifications from Alertmanager in
+# emails/sec. 0 = rate limit disabled. Negative value = no emails are allowed.
+# CLI flag: -alertmanager.email-notification-rate-limit
```

Comment:
Are you planning to have separate configuration for each notification type? I think there are some advantages to that in flexibility, but might make the config a bit complicated if this is extended to all the receivers.

Reply:
> Are you planning to have separate configuration for each notification type?

Yes, that was my plan. Do you suggest to use single rate-limit configuration for all notification types? We can also have one generic rate-limit config for all integrations, with per-integration-type overrides in case they are needed. WDYT?

Would you also suggest to use single rate-limiter shared across all notifiers? (I guess not, because eg. too many webhook notifications could stop email notifications).

Reply:
> We can also have one generic rate-limit config for all integrations, with per-integration-type overrides in case they are needed. 

Ah... we cannot easily distinguish between default values and missing values, so this would be difficult.

Reply:
> Ah... we cannot easily distinguish between default values and missing values, so this would be difficult.

Can we use pointer values in the config? If pointer is nil then it has not been set. I haven't checked if it's doable, just asking.

Reply:
> Can we use pointer values in the config? If pointer is nil then it has not been set. I haven't checked if it's doable, just asking.

This is doable, but could lead to confusing rulers when trying to setup overrides correctly.

Eg.:

global limits:
  - no defined shared (for all integrations) rate limit (a)
  - but defined email rate limit (b)

user A:
  - defined shared (for all integrations) rate limit  (c)
  - undefined email rate limit  (d)

Now when computing rate limit for email integration for "user A", we can either use his shared limits (c), or global email rate limits (b), but it's unclear what is a better option, and no matter what we choose, we will confuse some people.

I would leave that to a separate discussion and PR for now.

Reply:
> I would leave that to a separate discussion and PR for now.

I agree on this. @ranton256 What do you think? Some feedback on this would be great. Thanks!